libs/ledger-live-common/src/families/cosmos/api/Cosmos.ts

```diff
@@ -270,4 +270,20 @@ export class CosmosAPI {
 
     return patchOperationWithHash(operation, data.tx_response.txhash);
   };
+
+  simulate = async (tx_bytes: number[]): Promise<BigNumber> => {
+    try {
+      const { data } = await network({
+        method: "POST",
+        url: `${this.defaultEndpoint}/cosmos/tx/${this.version}/simulate`,
+        data: {
+          tx_bytes,
+        },
+      });
+
+      return new BigNumber(data?.gas_info?.gas_used || 0);
+    } catch (e) {
+      return new BigNumber(0);
```

Comment:
Naïve question, are we sure we want to default to 0?
Asking because sometimes default values are sneakier and worse than errors.

Reply:
I guess 0 is fine if we don't want the user to make the tx in that case
It depends if we want to allow the user to send with overpaid fees

Reply:
After discussing it with huiqi, if this returns 0, in the method on top of it we return the default gas value

Reply:
I think an error should be thrown, and then, on the top method if the error happens we use default value, wdyt ?

Reply:
Indeed, the way it's used now by the calling function looks OK so it won't change much, but in case simulate is used elsewhere someday, let's make it throw instead of returning 0, and update the tests accordingly :+1: